src/salte-auth.js

```diff
@@ -1,1226 +1,358 @@
- * @property {Array} anonymousEndpoints   Optional.  Array containing one or more endpoints to which a token need not be attached.  If the endpoint specified exists anywhere within the endpoint being called then it is considered to be a match.
- * @property {boolean} isAngular          Optional.  Used to tell this library whether it should handle the callback from the authorization server itself or bubble the response up to the main window's URL and let the Angular http interceptor handle it.
- * @property {function} displayCall       Optional.  If specified then this function will be called with the fully-formed authorization url when login is invoked.
- * @property {boolean} popUp              Optional; defaults to true.
- * @property {function} callback          Optional.  If specified then this function will be called upon receiving a successful response or error from the authorization server.
- * @property {string} hashPrefix          Optional; defaults to empty string.
- * @property {string} tokenCallbackTimeout Optional; defaults to 6 seconds.  This is the amount of time salte-auth will wait for the identity provider's token endpoint to respond.
+ * Disable certain security validations if your provider doesn't support them.
+ * @typedef {Object} Validation
+ * @property {Boolean} [nonce=true] Passing false will disable nonce validation, leaving you vulnerable to replay attacks.
+ * @property {Boolean} [state=true] Passing false will disable state validation, leaving you vulnerable to XSRF attacks.
+ * @property {Boolean} [azp=true] Passing false will disable azp validation.
+ * @property {Boolean} [aud=true] Passing false will disable aud validation.
  */
 
 /**
- * User information from idtoken.
- *  @class User
- *  @property {string} userName - username assigned from upn, email, or sub (i.e. subject).
- *  @property {object} profile - properties parsed from idtoken.
+ * The configuration for salte auth
+ * @typedef {Object} Config
+ * @property {String} gateway The base url of your identity provider.
+ * @property {('id_token'|'id_token token')} responseType The response type to authenticate with.
+ * @property {String} redirectUrl The redirect url specified in your identity provider.
+ * @property {String} clientId The client id of your identity provider
+ * @property {String} scope A list of space-delimited claims used to determine what user information is provided and what access is given. Most providers require 'openid'.
+ * @property {Boolean|Array<String>} routes A list of secured routes. If true is provided then all routes are secured.
```

Comment:
Do you think that routes makes it clear that this is referring to application navigation routes whereas endpoints refers to API endpoints?

Reply:
The term route is fairly common when referring to an applications routes.

**References**

- https://www.webcomponents.org/element/PolymerElements/app-route
- https://docs.angularjs.org/api/ngRoute/provider/$routeProvider
- https://github.com/ReactTraining/react-router
- https://router.vuejs.org/en/